airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -0,0 +1,268 @@
+"""
+MIT License
+
+Copyright (c) 2020 Airbyte
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterator, Sequence
+
+import pendulum
+from base_python.entrypoint import logger
+from facebook_business.adobjects.igmedia import IGMedia
+from facebook_business.exceptions import FacebookRequestError
+
+
+class StreamAPI(ABC):
+    result_return_limit = 100
+
+    def __init__(self, api, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._api = api
+
+    @abstractmethod
+    def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
+        """Iterate over entities"""
+
+
+class IncrementalStreamAPI(StreamAPI, ABC):
+    @property
+    @abstractmethod
+    def state_pk(self):
+        """Name of the field associated with the state"""
+
+    @property
+    def state(self):
+        return {self.state_pk: str(self._state)}
+
+    @state.setter
+    def state(self, value):
+        self._state = pendulum.parse(value[self.state_pk])
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = None
+
+    def state_filter(self, record: Dict) -> Iterator[Any]:
+        """Apply state filter to set of records, update cursor(state) if necessary in the end"""
+        cursor = pendulum.parse(record[self.state_pk])
+        stream_name = self.__class__.__name__
+        if stream_name.endswith("API"):
+            stream_name = stream_name[:-3]
+        logger.info(f"Advancing bookmark for {stream_name} stream from {self._state} to {cursor}")
+        self._state = max(cursor, self._state) if self._state else cursor
+        return record
+
+
+class IgUsersAPI(StreamAPI):
```

Comment:
"Ig" in beginning of each API name has not clear meaning. It should be either avoided or used full word "Instagram".

Reply:
lets also replace API word with Stream, StreamAPI -> Stream, UsersAPI -> UserStream (notice single, not plural),
wdyt?